zerver/lib/soft_deactivation.py

```diff
@@ -33,3 +37,24 @@ def do_soft_deactivate_users(users):
             )
             realm_logs.append(log)
         RealmAuditLog.objects.bulk_create(realm_logs)
+
+def get_users_for_soft_deactivation(realm, inactive_for_days):
+    # type: (Realm, int) -> List[UserProfile]
+    users_activity = list(UserActivity.objects.filter(
+        user_profile__realm=realm,
+        user_profile__is_bot=False,
+        user_profile__long_term_idle=False).values(
+        'user_profile_id').annotate(last_visit=Max('last_visit')))
+    user_ids_to_deactivate = []
+    today = timezone_now()
+    for user in users_activity:
+        if (today - user['last_visit']).days > inactive_for_days:
+            user_ids_to_deactivate.append(user['user_profile_id'])
```

Comment:
I'd call this local variable `user_activity`, not `user`; in the Zulip codebase, we try hard to have variable names suggest the type of the object; something called `user` or `user_profile` should be a UserProfile object.